File: backend/src/test/java/be/lutske/leolegacy/interfaceadapter/rest/PostgresIntegrationTestProfile.java
package be.lutske.leolegacy.interfaceadapter.rest;

import io.quarkus.test.junit.QuarkusTestProfile;

import java.util.Map;

/**
 * Quarkus test profile that uses the real PostgreSQL database (via podman-compose)
 * and real Tesseract OCR instead of H2 + mocked OCR.
 *
 * Prerequisite: {@code podman compose up -d} must be running.
 * Prerequisite: Tesseract must be installed ({@code brew install tesseract}).
 */
public class PostgresIntegrationTestProfile implements QuarkusTestProfile {

    @Override
    public Map<String, String> getConfigOverrides() {
        return Map.of(
                "quarkus.datasource.db-kind", "postgresql",
                "quarkus.datasource.username", "leo",
                "quarkus.datasource.password", "leo_secret",
                "quarkus.datasource.jdbc.url", "jdbc:postgresql://localhost:5432/leo_legacy",
                "quarkus.datasource.jdbc.acquisition-timeout", "5",
                "quarkus.flyway.migrate-at-start", "true",
                "quarkus.flyway.clean-at-start", "true",
                "ocr.tessdata-path", "/opt/homebrew/share/tessdata",
                "ocr.language", "eng"
        );
    }
}

File: backend/src/test/kotlin/be/lutske/leolegacy/interfaceadapter/rest/PostgresIntegrationTestProfile.kt
package be.lutske.leolegacy.interfaceadapter.rest

import io.quarkus.test.junit.QuarkusTestProfile

/**
 * Quarkus test profile that uses the real PostgreSQL database (via podman-compose)
 * and mocked AI extraction service.
 *
 * Prerequisite: `podman compose up -d` must be running.
 */
class PostgresIntegrationTestProfile : QuarkusTestProfile {

    override fun getConfigOverrides(): Map<String, String> {
        return mapOf(
            // Point to real PostgreSQL (same as compose.yaml)
            "quarkus.datasource.db-kind" to "postgresql",
            "quarkus.datasource.username" to "leo",
            "quarkus.datasource.password" to "leo_secret",
            "quarkus.datasource.jdbc.url" to "jdbc:postgresql://localhost:5432/leo_legacy",
            "quarkus.datasource.jdbc.acquisition-timeout" to "5",

            // Flyway: clean + migrate to get a fresh schema each test run
            "quarkus.flyway.migrate-at-start" to "true",
            "quarkus.flyway.clean-at-start" to "true",

            // AI config (extraction service is mocked in integration tests)
            "app.ai.provider" to "openai",
            "app.ai.model" to "gpt-4o-test",
            "app.ai.api-key" to "test-key-not-real",
        )
    }
}
